youtube_dl/extractor/rutube.py

```diff
@@ -17,7 +17,7 @@
 class RutubeIE(InfoExtractor):
     IE_NAME = 'rutube'
     IE_DESC = 'Rutube videos'
-    _VALID_URL = r'https?://rutube\.ru/(?:video|(?:play/)?embed)/(?P<id>[\da-z]{32})'
+    _VALID_URL = r'https?://rutube\.ru/(?:video|(?:play/)?embed)/(?P<id>[\da-z]{32})(/)?$'
```

Comment:
>There are several reasons why this strikes me as a non-optimal solution.

It's not broken as yours at least.

>we don't test for its characteristics

Write code so that it would test "for its characteristics" obviously.

>Adopting that approach may lead someone to add a different extractor in the future and our playlist extractor may match that specification, simply because its not defined properly

No, this won't happen due to `test_all_urls.py`. Conversely, usually distinguishing extractors with regexes only lead to breaking these tests.

>I don't see your approach solving much of a problem.

Write code that will `parse_qs` the query in `suitable` and check presence of required arguments. Solves it perfectly.